lib/cache.rb

```diff
@@ -0,0 +1,49 @@
+require 'sqlite3'
+require 'time'
+
+class Cache
+  DEFAULT_CACHE_EXPIRATION_TIME = 48 * 60 * 60
+
+  def self.create
+    @instance ||= new
+  end
+
+  def initialize
+    @db = SQLite3::Database.open 'CACHE.db'
+    @db.execute 'CREATE TABLE IF NOT EXISTS cache(key VARCHAR(16) PRIMARY KEY, expire_date TEXT, value TEXT)'
+  end
+
+  def read(key)
+    @db = SQLite3::Database.open 'CACHE.db'
```

Comment:
Vad ca 
1) nu inchizi deloc conexiunea
2) nu inchizi conexiunea la ResultSet (rezultatul la @db.query...) precum zic ei in docs https://github.com/sparklemotion/sqlite3-ruby/blob/194493be44cd41b73e612eed69ba49354f0d32e7/lib/sqlite3/database.rb#L317-L320

In Readme vad sa foloseasca doar execute, si nu gasesc teste pt query, dar ai putea face ceva de genul:

```
def read
  db = SQLite3::Database.open 'CACHE.db'
  results = db.query 'SELECT expire_date, value FROM cache WHERE key=?', key
  # do stuff
ensure
  results.close
  db.close
end
```

Reply:
Ideal ar fi sa ai un Connection ce se creeaza cand porneste aplicatia si se inchide cand opresti aplicatia.
Poate modificat CLI.run sau CommandHandler in genul:


```
def run
  Connection.init
  arguments = ArgumentsParser.parse(ARGV)
  result = CommandHandler.execute(arguments)
  ResultPrinter.print(result)
ensure
  Connection.close
end
```

si apoi unde ai avea conexiunea ai face

```
@db = Connection.instance
```

Dar trebuie sa fii atenta pt ca tot trebuie sa faci results.close pe ResultSet (returnat de metoda .query)

Reply:
SI trebuie sa mai adaugi si CACHE.db in .gitignore ca altfel trebuie sa ai mereu grija sa nu ii dai commit.

Reply:
Si poti face si ceva de genul

```
def read
  db = SQLite3::Database.open 'CACHE.db'
  db.query 'SELECT expire_date, value FROM cache WHERE key=?', key do |results|
    # do stuff
  end
ensure
  db.close
end
```

si zic ei ca fac ei automat close pt tine dupa ce se executa block-ul